EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,217 @@
+        private readonly ICharacterInventoryProvider _characterInventoryProvider;
+        private readonly IEOMessageBoxFactory _messageBoxFactory;
+        private readonly IEIFFileProvider _eifFileProvider;
+        private readonly ISfxPlayer _sfxPlayer;
+       
+        private CharacterRenderProperties RenderProperties => _characterControl.RenderProperties;
+        private ListDialogItem _changeHairItem, _changeHairColor, _changeBuyHairStyleOrColor;
+
+        public BarberDialog(INativeGraphicsManager nativeGraphicsManager,
+                            ICharacterRendererFactory rendererFactory,
+                            IEODialogButtonService dialogButtonService,
+                            ICharacterRepository characterLvRepository,
+                            IEODialogIconService dialogIconService,
+                            ILocalizedStringFinder localizedStringFinder,
+                            IBarberActions barberActions,
+                            ICharacterInventoryProvider characterInventoryProvider,
+                            IEOMessageBoxFactory messageBoxFactory,
+                            IEIFFileProvider eifFileProvider,
+                            ISfxPlayer sfxPlayer)
+            : base(nativeGraphicsManager, isInGame: true)
+        {
+            _characterRepository = characterLvRepository;
+            _dialogIconService = dialogIconService;
+            _localizedStringFinder = localizedStringFinder;
+            _barberActions = barberActions;
+            _characterInventoryProvider = characterInventoryProvider;
+            _messageBoxFactory = messageBoxFactory;
+            _eifFileProvider = eifFileProvider;
+            _sfxPlayer = sfxPlayer;
+
+            BackgroundTexture = GraphicsManager.TextureFromResource(GFXTypes.PostLoginUI, 56);
+            var mainCharacterRenderProperties = _characterRepository.MainCharacter.RenderProperties;
+            _characterControl = new CreateCharacterControl(mainCharacterRenderProperties, rendererFactory)
+            {
+                DrawPosition = new Vector2(210, 19)
+            };
+
+            InitializeCharacterControl();
+            InitializeDialogItems(dialogButtonService);
+            CenterInGameView();
+        }
+
+        private void InitializeCharacterControl()
+        {
+            _characterControl.SetParentControl(this);
+        }
+
+        private void InitializeDialogItems(IEODialogButtonService dialogButtonService)
+        {
+            var cancel = CreateButton(dialogButtonService, new Vector2(215, 150), SmallButton.Cancel);
+            cancel.OnClick += (_, _) => Close(XNADialogResult.Cancel);
+
+            CreateListItems();
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _characterControl.Initialize();
+            UpdateListItems(_characterRepository.MainCharacter.RenderProperties);
+        }
+
+        public void Update(GameTime gameTime)
```

Comment:
Remove this unused method (this also doesn't have `override` so it could cause issues too)